contracts/staking/l2_contracts/SymbioticStaking.sol

```diff
@@ -241,36 +223,34 @@ contract SymbioticStaking is
 
         _updateTxCountInfo(_numOfTxs, _captureTimestamp, SLASH_RESULT_TYPE);
 
-        // there could be no operator slashed
-        if (_jobSlashed.length > 0) IStakingManager(stakingManager).onSlashResult(_jobSlashed);
-
-        // TODO: unlock the selfStake and reward it to the transmitter
-        emit SlashResultSubmitted(msg.sender, _index, _numOfTxs, _imageId, _slashResultData, _proof);
+        // there could be no prover slashed
+        if (_taskSlashed.length > 0) IStakingManager(stakingManager).onSlashResultSubmission(_taskSlashed);
 
         // when all chunks of Snapshots are submitted
         if (_index == _numOfTxs - 1) {
```

Comment:
There doesn't seem to be any checks on index

Reply:
It's checked `_checkValidity`.
Did you mean anything else?